server/lib/chat/chat-api.ts

```diff
@@ -363,18 +365,19 @@ export class AdminChatApi {
       }),
     })
 
-    // TODO(2Pac): Move this to the chat-service
-    const channels = await searchChannelsAsAdmin({
+    return await this.chatService.searchChannels({
+      userId: ctx.session!.userId,
       limit,
       pageNumber: page,
       searchStr: searchQuery,
     })
-
-    return await this.chatService.getChannelInfos(
-      channels.map(c => c.id),
-      ctx.session!.userId,
-    )
   }
+}
+
+@httpApi('/admin/chat')
+@httpBeforeAll(ensureLoggedIn, convertChatServiceErrors)
+export class AdminChatApi {
+  constructor(private chatService: ChatService) {}
 
   @httpGet('/:channelId/messages')
   @httpBefore(checkAllPermissions('moderateChatChannels'))
```

Comment:
Think you should just put this on the class so we don't accidentally forget it one time, I can't see a use for any other permission here